src/node/db/PadManager.js

```diff
@@ -49,17 +49,18 @@ var globalPads = {
  * Updated without db access as new pads are created/old ones removed.
  */
 let padList = {
-  list: [],
+  list: new Set(),
+  cachedList: [],
   sorted : false,
   initiated: false,
   init: async function() {
-    let dbData = await db.findKeys("pad:*", "*:*:*");
+    let dbData = await db.findKeys("pad:*", "pad:*:*");
 
     if (dbData != null) {
       this.initiated = true;
 
       for (let val of dbData) {
-        this.addPad(val.replace(/pad:/,""), false);
+        this.addPad(val.replace(/^pad:/,""),false);
```

Comment:
This anchors the regexp to the start of the string. I see no dangers here, and a **performance gain**.
Nice catch.

This can be split in a dedicated commit.